### ISSUE TITLE
Memory problem between renderings

### ISSUE THREAD

**Issue by matugenos** [Haoming02/sd-webui-forge-classic#1275]
I have 16 gb VRAM and 16 gb system RAM. Using Forge Neo I can run Krea 2 and generate pictures with no problem (krea2_turbo_fp8) after I start Forge Neo but when my rendering is over (one or several pictures) if I try to render new pictures, the system RAM becomes saturated and hits the pagefile (virtual memory) for about 5 minutes before the rendering starts. It is much faster for me to completely close and restart Forge Neo when the rendering is finished than to hit "Render" again. This problem (system RAM saturation) never happens when it is my first rendering(s), only when the rendering is over and I try rendering again. There has to be a memory management problem between renderings because it works fine the first time. Somehow, when a rendering is over, Forge Neo doesn't revert to its pre-rendering state in terms of memory.

**Comment by chromesun** [comment]
After updating Neo on 17 July 2026, Krea 2 works for me - no longer getting the pagefile issue for 2nd and subsequent generations.

**Comment by matugenos** [comment]
Thanks a lot. I will try it. I have tried to narrow down this issue and it only happens when I change the intensity value of a LoRA. If I only change the prompt, there is no system RAM spike but if I only change the intensity value of a LoRA I get stuck in pagefile.

**Comment by matugenos** [comment]
I have upgraded to neo-2.27 but the problem is still there. 

After further testing, the system RAM memory spike only occurs when I change the intensity value of one specific LoRA called krea2filterbypass3.safetensors. The problem does not occur with other LoRAs. krea2filterbypass3 unlocks Krea2. It is a tiny LoRA with no memory footprint. To test the problem, do the following:

-generate an image using a prompt and <lora:krea2filterbypass3:5>
-generate an image using the same prompt and <lora:krea2filterbypass3:3>

It will cause a massive system RAM spike.

I use (16 gb VRAM + 16 gb system RAM PC):
-krea2_turbo_int8_convrot.safetensors (checkpoint)
-qwen3vl_4b_int8_convrot.safetensors (text encoder)
-Wan2_1_VAE_fp32.safetensors (VAE)
-Diffusion in Low Bits: Automatic
-set COMMANDLINE_ARGS= --xformers --sage --uv --cuda-malloc --cuda-stream
-With --pin-shared-memory or not the problem is the same

**Comment by chromesun** [comment]
I don't know how to diagnose the problem you are getting. Maybe change the bypass LoRA?
In case it helps, here is my working Krea 2 config.

I am using a "raw" Krea 2 because I found the "turbo" model too restrictive in aesthetics.
By adding a "Turbo" LoRA (see below for details) I do not have to use 50+ steps 🙂
Main: Euler/Simple, 8 steps.
Hiresfix: 4x_foolhardy_Remacri, Denoise 0.4, 8 steps.

PC: Ryzen 7 2700 (8c/16t), 64GB RAM, 12GB VRAM (RTX 3060/12).
NEO 2.27
set COMMANDLINE_ARGS=--uv --reserve-vram 2 --disable-sage --bnb --cuda-malloc
Diffusion in Low Bits: Automatic (fp16 LoRA)

The "--disable-sage" is because I was having black image problems with some models (not Krea 2).

I found the info here to be very useful, even though it is for Comfy:
https://www.reddit.com/r/StableDiffusion/comments/1ulxqep/krea_2_simple_gen_workflow_with_good_settings_for/
Neo does not have the samplers the author talks about ☹

The files I use are:

Model: krea2_raw-Q6_K (krea2_raw-Q6_K.gguf)
from: https://huggingface.co/vantagewithai/Krea-2-Raw-GGUF/tree/main
Module 1: Wan2_1_VAE_fp32 (Wan2_1_VAE_fp32.safetensors)
from: https://huggingface.co/Kijai/WanVideo_comfy/blob/main/Wan2_1_VAE_fp32.safetensors
Module 2: qwen3vl_4b_bf16 (qwen3vl_4b_bf16.safetensors)
from: https://huggingface.co/Comfy-Org/Krea-2/tree/main/text_encoders

I use these LoRAs with my Krea 2 RAW prompts:
Turbo Lora (krea2_turbo_lora_rank_64_bf16.safetensors) with strength 0.4 <---> 0.6
https://huggingface.co/Comfy-Org/Krea-2/blob/main/loras/krea2_turbo_lora_rank_64_bf16.safetensors
Anti-Censorship Lora (krea2filterbypass.safetensors) with strength=1 (it can go much higher)
https://civitai.com/models/2728234/krea2filterbypass?modelVersionId=3066812

**Comment by Haoming02** [comment]
Is this still a problem?

**Comment by chromesun** [comment]
It is intermittent for me - seems random. I can do several Krea2 gens, individual or batch, OK. Then one will go into swap. Then, usually, the next set of generations are fine. Sometimes they're not - but a Neo restart gets me back to successful generating. Happens, hmm, maybe 1 in 50. Not enough of an issue for me to shout about it. I am just happy to be able to use Krea2 the other 49 times 🙂
The other model I use most is ZIB - I never get the VRAM overload with it.
Comfy infuriates me, so I'm sticking with Neo.

**Comment by matugenos** [comment]
I only get this problem if I use the krea2filterbypass LoRA (version 1, 2 or 3 of the LoRA has the same problem). I have no problem with any other LoRA. My solution is to keep the intensity value of the LoRA always the same. But if I really have to change the value, I close and restart Forge Neo. I have not tried a version newer than neo-2.27 yet.

**Comment by chromesun** [comment]
Frequency of going into swap seems to be increasing. With latest git pull (0d0cb729..4bf0bf3b) I'm getting some oom as well. That's new for me. Windows Task Manager showed VRAM was about 10GB/12GB, and RAM was about 40GB/64GB. Console:
Encountered Out of Memory during Sampling; Unloading all Models...                                                                 main_thread.py :: ERROR
EDIT:
Error happened on 2nd image of a batch of 8. I set the batch to 1 and hit "Generate" again. Neo used swap instead of OOMing. The same job was OK yesterday (copied parameters from PNG info... Send to txt2img).
Console, starting from oom error above:

Encountered Out of Memory during Sampling; Unloading all Models...                                                                 main_thread.py :: ERROR
Loading Model: {'checkpoint_info': {'filename': 'G:\\stable-diffusion-webui-master\\webui\\models\\Stable-diffusion\\OTHER\\Krea2_Turbo_CyberRealistic_v2_int8_convrot.safetensors', 'hash': '30e6319b'}, 'additional_modules': ['C:\\FORGE\\webui\\models\\VAE\\Wan2_1_VAE_fp32.safetensors', 'C:\\FORGE\\webui\\models\\VAE\\qwen3-vl-4b-heretic_int8.safetensors'], 'unet_storage_dtype': None}
                                                                                                                                                            Using MixedPrecision for Qwen3                                                                                                          loader.py :: INFO
Using MixedPrecision for Model                                                                                                          loader.py :: INFO
Diffusion Model: {storage: torch.bfloat16, computation: Mixed}                                                                         k_model.py :: INFO
Model loaded in 2.0s (unload existing model: 0.3s, forge model load: 1.7s).
                                                                                                                                                            [LORA] Loaded The_Vividator_K2_e10.safetensors for KModel-UNet with 256 keys at weight 0.7 (skipped 0 keys) with on_the_fly = True    networks.py :: INFO
[LORA] Loaded krea2filterbypass.safetensors for KModel-UNet with 1 keys at weight 1.0 (skipped 0 keys) with on_the_fly = True         networks.py :: INFO
Negative Prompts are Ignored when CFG = 1.0                                                                                         processing.py :: INFO
Requested to load JointTextEncoder                                                                                           memory_management.py :: INFO
loaded completely; 8126.18 MB usable, 5359.03 MB loaded, full load: True                                                                  base.py :: INFO
Moving model(s) has taken 3.46 seconds                                                                                       memory_management.py :: INFO
Negative Prompts are Ignored when CFG = 1.0                                                                                         processing.py :: INFO
Requested to load KModel                                                                                                     memory_management.py :: INFO
loaded partially; 8125.61 MB usable, 8029.59 MB loaded, 4834.98 MB offloaded, 96.02 MB buffer reserved, lowvram patches: 0                base.py :: INFO
Moving model(s) has taken 8.04 seconds                                                                                       memory_management.py :: INFO
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [01:02<00:00,  6.30s/it]
Requested to load WanVAE                                                                                                     memory_management.py :: INFO
Unloaded partially: 3370.76 MB freed, 4658.82 MB remains loaded, 96.06 MB buffer reserved, lowvram patches: 0                             base.py :: INFO
loaded completely; 417.05 MB usable, 242.03 MB loaded, full load: True                                                                    base.py :: INFO
Moving model(s) has taken 1.61 seconds                                                                                       memory_management.py :: INFO
tiled upscale (GPU Composite): 100%|████████████████████████████████████████████████████████████████████████████████████████| 18/18 [00:06<00:00,  2.99it/s]
Unloaded partially: 242.03 MB freed, 0.00 MB remains loaded, 7.59 MB buffer reserved, lowvram patches: 0                                  base.py :: INFO
Moving model(s) has taken 2.38 seconds                                                                                       memory_management.py :: INFO
Requested to load KModel                                                                                                     memory_management.py :: INFO
loaded partially; 6362.07 MB usable, 6266.05 MB loaded, 6598.66 MB offloaded, 96.02 MB buffer reserved, lowvram patches: 0                base.py :: INFO
Moving model(s) has taken 1.75 seconds                                                                                       memory_management.py :: INFO
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [10:51<00:00, 81.46s/it]
Requested to load WanVAE                                                                                                     memory_management.py :: INFO
loaded partially; 0.00 MB usable, 0.00 MB loaded, 242.00 MB offloaded, 7.59 MB buffer reserved, lowvram patches: 0                        base.py :: INFO
Moving model(s) has taken 3.88 seconds                                                                                       memory_management.py :: INFO
Total progress: 50it [28:59, 34.79s/it]
Total progress: 50it [28:59, 115.57s/it]

Windows Task Manager screenshot taken at 4 of 8 in the Hires.fix process:

<img width="794" height="902" alt="Image" src="https://github.com/user-attachments/assets/8d13e912-afaa-4aa6-bdda-2b41e426c638" />

Models in use:
UI Preset	krea
Checkpoint	OTHER\Krea2_Turbo_CyberRealistic_v2_int8_convrot.safetensors
VAE / Text Encoder	Wan2_1_VAE_fp32.safetensors, qwen3-vl-4b-heretic_int8.safetensors
Diffusion in Low Bits	Automatic (fp16 LoRA)

Neo command line:
set COMMANDLINE_ARGS=--uv --port 7861 --reserve-vram 2 --disable-sage --cuda-malloc

Then, without doing anything else on the PC, I hit "Generate" again. Image generation completed with no problems:

Negative Prompts are Ignored when CFG = 1.0                                                                                         processing.py :: INFO
Requested to load JointTextEncoder                                                                                           memory_management.py :: INFO
loaded completely; 8126.49 MB usable, 5359.03 MB loaded, full load: True                                                                  base.py :: INFO
Moving model(s) has taken 2.45 seconds                                                                                       memory_management.py :: INFO
Negative Prompts are Ignored when CFG = 1.0                                                                                         processing.py :: INFO
Requested to load KModel                                                                                                     memory_management.py :: INFO
loaded partially; 8091.43 MB usable, 7995.41 MB loaded, 4868.72 MB offloaded, 96.02 MB buffer reserved, lowvram patches: 0                base.py :: INFO
Moving model(s) has taken 5.81 seconds                                                                                       memory_management.py :: INFO
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [01:06<00:00,  6.66s/it]
Requested to load WanVAE                                                                                                     memory_management.py :: INFO
Unloaded partially: 3336.58 MB freed, 4658.82 MB remains loaded, 96.06 MB buffer reserved, lowvram patches: 0                             base.py :: INFO
loaded completely; 382.87 MB usable, 242.03 MB loaded, full load: True                                                                    base.py :: INFO
Moving model(s) has taken 1.78 seconds                                                                                       memory_management.py :: INFO
tiled upscale (GPU Composite): 100%|████████████████████████████████████████████████████████████████████████████████████████| 18/18 [00:06<00:00,  2.86it/s]
Unloaded partially: 242.03 MB freed, 0.00 MB remains loaded, 7.59 MB buffer reserved, lowvram patches: 0                                  base.py :: INFO
Moving model(s) has taken 2.47 seconds                                                                                       memory_management.py :: INFO
Requested to load KModel                                                                                                     memory_management.py :: INFO
loaded partially; 6327.89 MB usable, 6231.87 MB loaded, 6632.41 MB offloaded, 96.02 MB buffer reserved, lowvram patches: 0                base.py :: INFO
Moving model(s) has taken 1.99 seconds                                                                                       memory_management.py :: INFO
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [04:52<00:00, 36.51s/it]
Requested to load WanVAE                                                                                                     memory_management.py :: INFO
loaded partially; 0.00 MB usable, 0.00 MB loaded, 242.00 MB offloaded, 7.59 MB buffer reserved, lowvram patches: 0                        base.py :: INFO
Moving model(s) has taken 4.48 seconds                                                                                       memory_management.py :: INFO
Total progress: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████| 18/18 [06:30<00:00, 21.71s/it]
Total progress: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████| 18/18 [06:30<00:00, 53.70s/it]

**Comment by danielfdo13-commits** [comment]
I can confirm I get this issue too when I'm using krea2filterbypass3 lora for 12GB+ Krea2 models. I have 16gb VRAM and 32GB system RAM. First few generations around ~20 seems to generate properly, then system RAM spikes and reaches 99% and hits the page file I think. I have to restart forge neo again to resolve this.

**Comment by Haoming02** [comment]
Please try again on the latest commit~

**Comment by chromesun** [comment]
Unfortunately, it's worse with the latest build (efc42fe).
My system is Ryzen 7-2700 (8c/16t), 64GB sysram, RTX 3060/12, SSDs for Windows 10 and model storage.
All my usage at the moment is Krea 2.

With my normal startup of:
set COMMANDLINE_ARGS=--uv --port 7861 --reserve-vram 2 --disable-sage --cuda-malloc
it's now maybe 1 in 4 gens that go into "Shared GPU memory" (sorry, I've been using "swap" when actually TaskMan says "shared").
Sometimes it's the first gen after starting Neo, sometimes I can get a batch of 4 done with no Shared usage.

I tried changing commandline to:
set COMMANDLINE_ARGS=--uv --port 7861 --reserve-vram 3 --disable-sage --cuda-malloc --cuda-stream --pin-shared-memory
but no change.
I thought the "--reserve-vram 3" might make things better, but no.

Typically I try 1280x720 when developing a prompt, changing to 1280x720 *2 Hires.fix = 2560x1440 when I'm happy. The 1280x720 images all generate fine - the problem is when I'm using Hires.fix.

Other main workflow is tryouts at 720x720, then full images at 720x720 * 2 Hires.fix = 1440x1440. I still get the overflow, but not as often.
Since I'm using Krea 2, I've also tried **without** Hires.fix: generating at 1440x1440 and 2048x1152. Again, overflows but not as often as the 2560x1440 path above.

I keep trying to find a deterministic scenario, but no luck so far. A frustrating thing is that using the PC for other things while generating seems to make no difference (watching dl'd Youtubes, listening to music). Earlier I just sat and watched the console+Taskman while a batch of 4 2560x1440 images were being generated. Images 1, 2 and 3 took about 4.5 minutes each. Image 4 took about 23 minutes. Pressed Generate again - all 4 were 4.5 minutes. Pressed Generate again, and image 1 took about 13 minutes. Aborted the batch there.

All my generating today has been with:
12.57GB	CyberRealistic_Krea2_V2.0_BF16_int8_convrot.safetensors
https://civitai.red/models/2831028/cyberrealistic-krea-2?modelVersionId=3225443
484MB	Wan2_1_VAE_fp32 (Wan2_1_VAE_fp32.safetensors)
https://huggingface.co/Kijai/WanVideo_comfy/blob/main/Wan2_1_VAE_fp32.safetensors
4.84GB	qwen3-vl-4b-heretic_int8.safetensors
from: https://huggingface.co/DreamFast/Qwen3-VL-4b-Heretic-ComfyUI/tree/main

Anti-Censorship Lora (krea2filterbypass.safetensors) 
https://civitai.com/models/2728234/krea2filterbypass?modelVersionId=3066812
And my own LoRA - trained on Civitai with default settings, not published, but it's simply a Krea 2 version of my Flux LoRA here:
https://civitai.red/models/2176911/the-vividator

I see that the Neo console is now telling me:
"You are running PyTorch 2.11.0+cu130, which is outdated."
so I will delete my venv and do a reinstall.

**Comment by chromesun** [comment]
Venv recreated.
Running gens that do 8 step 1280x270 with 8 step hires.fix = 2. Seems to be a 50/50 chance of VRAM spilling into Shared at hires step 5.

BUT: When I change hires.fix steps to 4, VRAM **never** spills into Shared.

Detail loss in resulting images seems minimal.
